### ISSUE TITLE
Adopt HYSOPTs BCM in IDEAS

### ISSUE THREAD

**Issue by rubenbaetens** [open-ideas/IDEAS#163]
I would (strongly) suggest to consider adopting ```HYSOPTs``` Base Circuit Methodology in ```IDEAS``` (which similar to the MODULEs in ```EES```). More info on this can be found in < R. Vandenbulcke (2013). Optimalisatie van hydronische verwarmingsinstallaties door middel van simulatie. PhD Thesis. >

**Comment by arnoutaertgeerts** [comment]
@rubenbaetens If I'm the assignee, I'm responsible for the merge? What needs to be checked before we can merge?

**Comment by rubenbaetens** [comment]
Arnout, you're the assignee from the Issue, not from the Pull Request ;) And those are preferably two different persons.

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts when using it, it seems the base circuits are by far not yet ready to be merged, as a majority of the necessary parameters are not propagated. i'm working on it.

**Comment by arnoutaertgeerts** [comment]
I was unable to use the ```inHomeGrid``` model on this branch due to two errors of dividing by zero in ```IDEAS.Electric.Distribution.BaseClasses.Branch.prescribedHeatFlow```. I "solved" both errors by adjusting these parameters:

- ```prescribedHeatFlow.alpha = 0``` => 1
- ``` Plos = R*Modelica.ComplexMath.'abs'(i)^2``` => + 1

Did anyone encounter these errors as well? I had to change these things in order for my building models to run.

**Comment by rubenbaetens** [comment]
Both changes imply the statement that the Ohm's law is incorrect; So i suggest reverting these changes and that we look for the cause instead of a solution ;)

**Comment by arnoutaertgeerts** [comment]
Ok, That might be a better solution :) I just had to quickly remedy these
errors in order for my models to run :)

**Comment by rubenbaetens** [comment]
If you "just had to quickly remedy these errors for me" i don't want to see them here in a commit, especially if there physically wrong ;)

**Comment by arnoutaertgeerts** [comment]
ok boss :)

**Comment by arnoutaertgeerts** [comment]
Did a big update of the basecircuit models with a complete restructuring of the partials. Could you guys @rubenbaetens  and @Mathadon take a look?

**Comment by arnoutaertgeerts** [comment]
Done in https://github.com/open-ideas/IDEAS/pull/230

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts i get an error when checking ```IDEAS.Fluid.BaseCircuits.Interfaces.PartialValveCircuit``` saying
```
*** Class or component 'rhoStdTot' not found in valveParametersTop
```
though i don't get why. Do you see it ?

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts found it: you wrote ```rhoStdTot``` instead of ```rhoStdTop```.

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts i get an error when checking ```BaseCircuits.Interfaces.PartialPumpCircuit``` saying
```
*** Class or component 'motorCooledByFluid', 'motorEfficiency' and 'hydraulicEfficiency' not found in PartialFlowmachine
```
Do these propagations need to be deleted ?

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts i solved the ```rhoStdTot``` and ```ThermostaticValve``` error, the last one i don't really get.

**Comment by rubenbaetens** [comment]
@arnoutaertgeerts it seems there is an error in ```BaseClasses.PumpSupply_dp``` as it is not balanced when you *Check* the component

**Comment by rubenbaetens** [comment]
Closed in #233
